auth/mtls/mtls.go

```diff
@@ -55,6 +58,70 @@ type CredentialsLoader interface {
 	// LoadServerCertificate returns the certificate that should be presented
 	// by the server to incoming clients.
 	LoadServerCertificate(context.Context) (tls.Certificate, error)
+
+	// CertRefreshed indicates if internally any of the cert data has
+	// been refreshed and should be reloaded. This will depend on the
+	// implementation to support but allows for dynamic refresh of certificates
```

Comment:
I suppose either the implementation will need to do work (including maybe file I/O) in this call context, or have an internal go-routine to set an 'update' flag. I wonder if we want to make some sort of implementation recommendation here?

Reply:
Yeah...the credential loaders we provide just return false always so this never happens. It really is context driven i.e. I mocked up a vault PR based on this and it was exactly that where we track and updated state and clear it once it gets checked.

I'm not sure what we could recommend here without just providing something like vault as an example?